lib/detect/mraid.js

```diff
@@ -47,5 +48,72 @@ module.exports = {
 		} else {
 			return '-1';
 		}
-	}
+	},
+
+    /**
+     *
+     * @param version
+     * @returns {{version: null||String, issues: Array}}
+     */
+	diagnostic: function (win) {
+		win = win || window;
+
+        var report = {
+            version: null,
+            issues: []
+        },
+            FUNCTION = 'function',
+            mraidObj = has('mraid', win);
+
+        var mraid1Props = {
+            addEventListener: FUNCTION,
+            close: FUNCTION,
+            expand: FUNCTION,
+            getExpandProperties: FUNCTION,
+            getPlacementType: FUNCTION,
+            getState: FUNCTION,
+            getVersion: FUNCTION,
+            isViewable: FUNCTION,
+            open: FUNCTION,
+            removeEventListener: FUNCTION,
+            setExpandProperties: FUNCTION,
+            useCustomClose: FUNCTION
+        };
+
+        var mraid2Props = {
+            createCalendarEvent: FUNCTION,
+            getCurrentPosition: FUNCTION,
+            getDefaultPosition: FUNCTION,
+            getMaxSize: FUNCTION,
+            getResizeProperties: FUNCTION,
+            getScreenSize: FUNCTION,
+            playVideo: FUNCTION,
+            setResizeProperties: FUNCTION,
+            storePicture: FUNCTION,
+            supports: FUNCTION
+        };
+
+        if (!mraidObj) {
+            report.issues.push('Missing MRAID');
+            return report;
+        }
+
+        report.version = mraidObj.getVersion() || '-1';
```

Comment:
This should be updated to check that 'getVersion' exists and is a function before calling it for the unlikely but possible case where the 'mraidObj' is not null/undefined because the 'window.mraid' variable exists but does not have a 'getVersion' method.